Lobby/EOSLobby.cs

```diff
@@ -0,0 +1,197 @@
+﻿using EpicTransport;
+using Epic.OnlineServices.Lobby;
+using UnityEngine;
+using Epic.OnlineServices;
+using System.Collections.Generic;
+
+public class EOSLobby : MonoBehaviour
+{
+    [HideInInspector] public bool ConnectedToLobby = false;
+    public LobbyDetails ConnectedLobbyDetails = new LobbyDetails();
+    
+    [SerializeField] public string[] AttributeKeys = new string[0];
+
+    public static string DefaultAttributeKey = "default";
+    private static string hostAddressKey = "host_address";
+
+    private string currentLobbyId = string.Empty;
+    private bool isLobbyOwner = false;
+
+    //events
+    public delegate void JoinLobbyHandler(List<Attribute> attributes);
+    public event JoinLobbyHandler JoinLobbyComplete;
+
+    public delegate void FindLobbiesHandler(List<LobbyDetails> foundLobbies);
+    public event FindLobbiesHandler FindLobbiesComplete;
+
+    //creates a lobby
+    public virtual void CreateLobby(uint maxConnections, LobbyPermissionLevel permissionLevel, bool presenceEnabled, AttributeData[] lobbyData = null)
+    {
+        EOSSDKComponent.GetLobbyInterface().CreateLobby(new CreateLobbyOptions 
+        { 
+            //lobby options
+            LocalUserId = EOSSDKComponent.LocalUserProductId, 
+            MaxLobbyMembers = maxConnections,
+            PermissionLevel = permissionLevel,
+            PresenceEnabled = presenceEnabled,
+        }, null, (CreateLobbyCallbackInfo callback) => 
+        {
+            List<Attribute> lobbyReturnData = new List<Attribute>();
+
+            //if the result of CreateLobby is not successful, log and error and return
+            if(callback.ResultCode != Result.Success)
+            {
+                Debug.LogError("There was an error while creating a lobby. Error: " + callback.ResultCode);
```

Comment:
We should provide an event for on join lobby failed as the ui probably needs that info

Reply:
I will add this for every method.